tests/test_workspacebuilder.py

```diff
@@ -1183,3 +1183,37 @@ def f():
 
         # handle case with OS X adding /private/ to /tmp/ paths
         assert retry_until(f)
+
+
+@pytest.mark.skipif(
+    has_lt_version("2.9"), reason="needs option introduced in tmux >= 2.9"
+)
+def test_layout_main_horizontal(session):
+    yaml_config = test_utils.read_config_file("workspacebuilder/three_pane.yaml")
+
+    sconfig = kaptan.Kaptan(handler="yaml")
+    sconfig = sconfig.import_config(yaml_config).get()
+
+    builder = WorkspaceBuilder(sconf=sconfig)
+    builder.build(session=session)
+
+    assert session.windows
+    window = session.windows[0]
+
+    assert len(window.panes) == 3
+    main_horizontal_pane, *panes = window.panes
+
+    def height(p):
+        return int(p._info["pane_height"])
+
+    def width(p):
+        return int(p._info["pane_width"])
+
+    assert height(main_horizontal_pane) > height(panes[0])
```

Comment:
I learned something new:

`main-horizontal` doesn't necessarily guarantee the top pane will be bigger.  

> A large (main) pane is shown at the top of the window and the remaining panes are spread from left to right in the leftover space at the bottom. Use the main-pane-height window option to specify the height of the top pane. [manpage](https://manpages.debian.org/buster/tmux/tmux.1.en.html#main-horizontal)

I will try to remedy this in #809